### PR TITLE
[4000] Show Loader instead of Empty Cart on reorder

### PR DESCRIPTION
**Pull request by leonazava**
**Related issue(s):**
* Fixes scandipwa/scandipwa/issues/4000

**Problem:**
* After pressing "Reorder", the cart displayed "Cart is empty" instead of a loader

**In this PR:**
* Changed the loader display condition from something that never happens (the "items" prop is always at least an empty array)
* Added the isLoading state and made its value to be the condition on which the loader renders 
* Updated Cart.dispatcher.js to set isLoading to true/false, previously it was always false

### CODE REVIEW

File: packages/scandipwa/src/route/CartPage/CartPage.component.js [hunk 1]
```diff
@@ -45,6 +45,8 @@ export class CartPage extends PureComponent {
         onCartItemLoading: null
     };
 
+    initial = true;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/CartPage/CartPage.component.js:48]: Use component state

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Comment by aleksandrakorolova** [comment]
Loader is added
1. Please, show previously added products in the cart + loader when adding products from order
https://drive.google.com/file/d/1WqfILa23Q18BA6jvUbE8JoiK8cs4C38p/view?usp=sharing
2. Loader is displayed not in the center of page if user didn't have products in the cart
Screenshot 2022-03-22 at 11 52 22 (2)

**Comment by aleksandrakorolova** [comment]
1. Fixed
2. Fixed
3. Mobile: User sees 2 loaders when doing reorder if cart was empty.
Screenshot 2022-03-24 at 08 11 10
4. Mobile: User doesn't see loader if user has a lot of products in the cart. Show loader within screen
Screenshot 2022-03-24 at 08 10 55

**Comment by aleksandrakorolova** [comment]
2. Loader is displayed not in the center of page if user didn't have products in the cart
Screenshot 2022-03-25 at 08 13 09 (2)
3. Fixed
4. Fixed
5. Loader is shifted down when doing reorder if cart was empty or refresh page
Screenshot 2022-03-25 at 08 29 38
6. Two loaders are displayed when doing reorder and user already has 1-2 products in the card
Screenshot 2022-03-25 at 08 29 49

**Comment by leonazava** [comment]
Couldn't replicate 5 and 6, but I expect this new commit should patch them as well.

**Pull request closed without merging**